x-pack/plugins/cloud_security_posture/public/test/test_provider.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { I18nProvider } from '@kbn/i18n-react';
+import { Router, Switch, Route } from 'react-router-dom';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { coreMock } from '../../../../../src/core/public/mocks';
+import { KibanaContextProvider } from '../../../../../src/plugins/kibana_react/public';
+import type { CspAppDeps } from '../application/app';
+
+const queryClient = new QueryClient();
+const defaultCore = coreMock.createStart();
+const defaultParams = coreMock.createAppMountParameters();
+
+export const TestProvider: React.FC<Partial<CspAppDeps>> = ({
+  core = defaultCore,
+  deps = {},
+  params = defaultParams,
+  children,
+} = {}) => {
+  return (
+    <KibanaContextProvider services={{ ...deps, ...core }}>
```

Comment:
```suggestion
export const TestProvider: React.FC<Partial<CspAppDeps>> = ({
  core = coreMock.createStart(),
  deps = {},
  params = coreMock.createAppMountParameters(),
  children,
} = {}) => {
  const queryClient = useMemo(() => new QueryClient(), []);
  return (
    <KibanaContextProvider services={{ ...deps, ...core }}>"+y
```

The values `queryClient`, `defaultCore`, and `defaultParams` could be mutated by test runs, but as it is, they aren't reset between test runs. By re-instantiating them for each instantiation of `TestProvider`, you could help ensure that tests runs are isolated.